libminifi/test/sql-tests/mocks/MockConnectors.cpp

```diff
@@ -0,0 +1,386 @@
+  } else if (minifi::utils::StringUtils::startsWith(query, "select")) {
+    return select(query, args);
+  } else {
+    throw std::runtime_error("Unknown query type");
+  }
+
+  return nullptr;
+}
+
+void MockDB::createTable(const std::string& query) {
+  std::smatch match;
+  std::regex expr("create table (\\w+)\\s*\\((.*)\\);");
+  std::regex_search(query, match, expr);
+  std::string table_name = match[1];
+  auto columns_with_type = minifi::utils::StringUtils::splitAndTrim(match[2], ",");
+  std::vector<std::string> col_names;
+  std::vector<DataType> col_types;
+  for (const auto& col_with_type : columns_with_type) {
+    auto splitted = minifi::utils::StringUtils::splitAndTrim(col_with_type, " ");
+    col_names.push_back(splitted[0]);
+    col_types.push_back(stringToDataType(splitted[1]));
+  }
+  tables_.emplace(table_name, MockRowset{col_names, col_types});
+  storeDb();
+}
+
+void MockDB::insertInto(const std::string& query, const std::vector<std::string>& args) {
+  std::string replaced_query = query;
+  for (const auto& arg : args) {
+    replaced_query = minifi::utils::StringUtils::replaceOne(replaced_query, "?", arg);
+  }
+
+  std::smatch match;
+  std::regex expr("insert into (\\w+)\\s*(\\((.*)\\))*\\s*values\\s*\\((.+)\\)");
+  std::regex_search(replaced_query, match, expr);
+  std::string table_name = match[1];
+  std::vector<std::string> values = minifi::utils::StringUtils::splitAndTrim(match[4], ",");
+  for (auto& value : values) {
+    value = minifi::utils::StringUtils::removeFramingCharacters(value, '\'');
+  }
+  auto insert_col_names = minifi::utils::StringUtils::splitAndTrim(match[3], ",");
+  if (!insert_col_names.empty()) {
+    auto col_names = tables_.at(table_name).getColumnNames();
+    std::vector<std::string> row;
+    for (const auto& col_name : col_names) {
+      auto it = std::find(insert_col_names.begin(),  insert_col_names.end(), col_name);
+      if (it != insert_col_names.end()) {
+        row.push_back(values.at(it-insert_col_names.begin()));
+      } else {
+        row.push_back("NULL");
+      }
+    }
+    tables_.at(table_name).addRow(row);
+  } else {
+    tables_.at(table_name).addRow(values);
+  }
+
+  storeDb();
+}
+
+std::unique_ptr<Rowset> MockDB::select(const std::string& query, const std::vector<std::string>& args) {
+  std::string replaced_query = query;
+  for (const auto& arg : args) {
+    replaced_query = minifi::utils::StringUtils::replaceOne(replaced_query, "?", arg);
+  }
+
+  std::smatch match;
+  std::regex expr("select\\s+(.+)\\s+from\\s+(\\w+)\\s*(where ((.+(?= order by))|.+$))*\\s*(order by (.+))*");
+  std::regex_search(replaced_query, match, expr);
+  auto cols = minifi::utils::StringUtils::splitAndTrim(match[1], ",");
+  if (cols[0] == "*") {
+    cols = {};
+  }
+  std::string table_name = match[2];
+  std::string condition_str = match[4];
+  std::function<bool(const MockRow&)> condition;
+  if (!condition_str.empty()) {
+    if (condition_str == "int_col > 103") {
```

Comment:
I feel like we should go for a different solution than hard-coded conditions, I agree that writing an expression parser would complicate things a lot, could we reuse some expression parsing utility from our codebase? (e.g. from Expression.cpp)

Reply:
Two problems with reusing Expression.cpp:
1. It's not available when compiling with expression language support disabled
2. It's parsing nifi expression language, but this looks like SQL to me

Not sure what would be the best way to solve this.

Reply:
maybe we could cover our test use-cases with some well-crafted regex-es (and maybe calling further regex-es on submatches if need be)

Reply:
I'll investigate some simple parsing solutions for handling our current cases. I think we need to find a good balance between maintainability and expandability. I used a hard coded condition as a start because we currently do not have too many conditions and it can be expanded easily, but if we add way more then it will become bloated. On the other side if we write a complex parser the maintainability would be way too much effort. I'll try to write a simple parser that would be sufficient for our cases without too much maintenance costs.

Reply:
Added a parser for simple where conditions in 96c3a9dea8e0d9c0492ccf5850184755651418e9

Reply:
a comment might come in handy, that it parses an `and`-separated list of conditions in the format of `<column_name> [<>=] <value>`

Reply:
Added comment in 182d8a534045a230857d7c5dc5116466f5177bcf